File: python/src/python/com/example/script/side1.py
from m8test_java.com.m8test.script.GlobalVariables import _logger
from m8test_java.com.m8test.script.GlobalVariables import _events
from m8test_java.com.m8test.script.GlobalVariables import _script

_logger.info("PYTHON_SIDE1_STARTED")

# 获取事件订阅器，用于订阅事件
subscriber = _events.getSubscriber()
# 当前事件 API 通过 scope selector + channel 标识订阅范围。
subscriber.subscribe(lambda scopes: scopes.getEngine(), "subscription-channel",
                      lambda event: _logger.info(
                          "收到事件 data=" + str(event.getPayload().getStringOrNull("data"))
                          + " time=" + str(event.getTimeMillis())
                          + " channel=" + event.getChannel()))
subscriber.subscribe(lambda scopes: scopes.getApp(), "subscription-channel",
                      lambda event: _logger.info(
                          "收到全局事件 data=" + str(event.getPayload().getStringOrNull("data"))
                          + " time=" + str(event.getTimeMillis())
                          + " channel=" + event.getChannel()))
# 获取脚本主线程
mainThread = _script.getThreads().getMain()
# 获取定时器
timer = mainThread.getTimer()
# 添加一个延时任务，为了防止脚本过早结束
timer.setTimeout(lambda it: it
                 # 保持脚本3秒钟不结束
                 , 3000)

File: php/src/com/example/script/side1.php
<?php
/** @var m8test_java\com\m8test\script\core\api\console\Console $console */
/** @var m8test_java\com\m8test\script\core\api\event\Events $events */
$console->log("PHP side1 started");
// 获取事件订阅器，用于订阅事件
$subscriber = $events->getSubscriber();
// 同一个 Engine 中的脚本使用 engine scope 互通事件。
$subscriber->subscribe(function ($scopes) {
    return $scopes->getEngine();
}, "subscription-channel", function ($event) use ($console) {
    $console->log("PHP side1 received event");
});
/** @var m8test_java\com\m8test\script\core\api\engine\Script $script */
// 获取脚本主线程
$mainThread = $script->getThreads()->getMain();
// 获取定时器
$timer = $mainThread->getTimer();
// 添加一个延时任务，为了防止脚本过早结束
$timer->setTimeout(function ($it) {
    // 保持脚本3秒钟不结束
}, 3000);
